### PR TITLE
chore: Remove ApiMayChange on Projection and RES over gRPC

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala [hunk 1]
```diff
@@ -556,7 +556,7 @@ private[projection] class R2dbcOffsetStore(
     val seqNr = recordWithOffset.record.seqNr
     val currentState = getState()
 
-    val duplicate = getState().isDuplicate(recordWithOffset.record)
+    val duplicate = currentState.isDuplicate(recordWithOffset.record)
```
Comment (patriknw) [inline akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala:559]: this is unrelated, small cleanup that I had lying around

**Review by octonato — APPROVED** [review]
LGTM

File: akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/ConsumerFilter.scala [hunk 2]
```diff
@@ -592,7 +590,6 @@ object ConsumerFilter extends ExtensionId[ConsumerFilter] {
 
 }
 
-@ApiMayChange
 class ConsumerFilter(system: ActorSystem[_]) extends Extension {
```
Comment (johanandren) [inline akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/ConsumerFilter.scala:593]: Should maybe be final since not meant for user extension

File: akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/javadsl/GrpcReadJournal.scala [hunk 3]
```diff
@@ -83,7 +81,6 @@ object GrpcReadJournal {
 
 }
 
-@ApiMayChange
 class GrpcReadJournal(delegate: scaladsl.GrpcReadJournal)
```
Comment (johanandren) [inline akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/javadsl/GrpcReadJournal.scala:84]: Final since public and not meant for extension

File: akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/javadsl/ReplicatedBehaviors.scala [hunk 4]
```diff
@@ -17,7 +16,6 @@ import akka.persistence.typed.javadsl.ReplicationContext
  *
  * Can optionally be composed with other Behavior factories, to get access to actor context or timers.
  */
-@ApiMayChange
 abstract class ReplicatedBehaviors[Command, Event, State] {
```
Comment (johanandren) [inline akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/javadsl/ReplicatedBehaviors.scala:19]:
```
* Not for user extension
*/
@DoNotInherit
```

Both language versions, we implement it and hand it to the user who only uses it

File: akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/scaladsl/Replica.scala [hunk 5]
```diff
@@ -23,7 +22,6 @@ object Replica {
 
 }
 
-@ApiMayChange
 trait Replica {
```
Comment (johanandren) [inline akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/scaladsl/Replica.scala:25]:
```suggestion
/**
 * Not for user extension
 */
@DoNotInherit
trait Replica {
```

File: akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/scaladsl/ReplicationProjectionProvider.scala [hunk 6]
```diff
@@ -19,7 +18,6 @@ import akka.stream.scaladsl.FlowWithContext
 /**
  * Factory/function for creating the projection where offsets are kept track of for the replication streams
  */
-@ApiMayChange
 trait ReplicationProjectionProvider {
```
Comment (johanandren) [inline akka-projection-grpc/src/main/scala/akka/projection/grpc/replication/scaladsl/ReplicationProjectionProvider.scala:21]:
```suggestion
@FunctionalInterface
trait ReplicationProjectionProvider {
```

**Review by johanandren — APPROVED** [review]
LGTM with some smaller stuff

**Comment by patriknw** [comment]
thanks, incorporated feedback

**Review by johanandren — APPROVED** [review]
